repository_test.go

```diff
@@ -577,6 +578,23 @@ func (s *RepositorySuite) TestPlainCloneContext(c *C) {
 	c.Assert(err, NotNil)
 }
 
+func (s *RepositorySuite) TestPlainCloneContextWithIncorrectRepo(c *C) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	tmpDir := c.MkDir()
+	repoDir := path.Join(path.Dir(tmpDir), "repoDir")
+	r, err := PlainCloneContext(ctx, repoDir, false, &CloneOptions{
+		URL: "incorrectOnPurpose",
+	})
+	c.Assert(r, IsNil)
+	c.Assert(err, NotNil)
+
+	_, err = os.Stat(repoDir)
+	dirNotExist := os.IsNotExist(err)
```

Comment:
There should be another test to check that if the directory exists, it is not removed by PlainClone.